File: 566.reshape-the-matrix.py
#
# @lc app=leetcode id=566 lang=python3
#
# [566] Reshape the Matrix
#

# @lc code=start
import math
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        flat = [value for row in mat for value in row]
        if r * c != len(flat):
            return mat

        new = []
        for i in range(0, len(flat), c):
            new.append(flat[i:i+c])
        return new
# @lc code=end

File: js/566.reshape-the-matrix.js
/*
 * @lc app=leetcode id=566 lang=javascript
 *
 * [566] Reshape the Matrix
 */

// @lc code=start
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(mat, r, c) {
  if (mat.length * mat[0].length !== r * c)
    return mat

  let flat = mat.reduce((a,b)=>{
    return [...a, ...b]
  }, [])

  let result = []
  for (let cr=0; cr < r; cr++) {
    result.push(flat.splice(0,c))
  }
  return result
};
// @lc code=end

